### ISSUE TITLE
NO bounding boxes

### ISSUE THREAD

**Issue by jaideep11061982** [experiencor/keras-yolo2#260]
Hey 
I tired to fit your code to my image  attached is one such.Followed below steps

1) I used only two classes so gave labels Date,NoDate instead of objects name.
2) I used BoundingDraw tool available in gitbub using 
https://github.com/puzzledqs/BBox-Label-Tool
3) Then I loaded the images sample output of image list and label list was as follows
[{'object': [{'name': 'NoDate', 'xmin': 0, 'xmax': 0, 'ymin': 0, 'ymax': 0}], 'filename': 'C:\\work\\machine learning\\project\\images1\\receipt1.jpeg', 'width': 380, 'height': 380}, {'object': [{'name': 'Date', 'xmin': 1, 'xmax': 187, 'ymin': 214, 'ymax': 234}], 'filename': 'C:\\work\\machine learning\\project\\images1\\receipt2.jpeg', 'width': 380, 'height': 380}]
Seen labels:	 {'NoDate': 12, 'Date': 16}

4) I trained it using Inception 3 .Below are the results of last epoch and last step

Loss XY         [0]
Loss WH         [0]
Loss Conf       [0.0213039108]
Loss Class      [0]
Total Loss      [0.0213039108]
Current Recall  [0]
Average Recall  [0]

5) then I finally tried to predict using below code
LABELS=['Date','NoDate']
image = cv2.imread('images1/receipt4.jpeg')
print(yolo.predict(image))
img=draw_boxes(image, yolo.predict(image), labels=LABELS)
plt.imshow(img[:,:,::-1])
plt.show()

6) In the images I did not get any bounding box  
where m i missing...
![receipt4](https://user-images.githubusercontent.com/38164196/39095235-bea2e010-465a-11e8-9352-66dcdc6d5ee4.jpeg)

**Comment by experiencor** [comment]
What do you want to detect here? The location of the date?

**Comment by jaideep11061982** [comment]
yes i want BB to be drawn around the date
I m not sure if i put the labels in the cfg correctly or not . I put it based on the distinct classes i have got which is just a binary here

**Comment by experiencor** [comment]
I think you don't need nodate bboxes.

**Comment by jaideep11061982** [comment]
also importantly the how should be anotation xml  in case of no date in image.
If i dont put the tag bndbox then i end up with key error exception in aug_image method of class YOLO.

<object>
		<name>NoDate</name>
		<pose>Frontal</pose>
		<truncated>0</truncated>
		<difficult>0</difficult>
		<occluded>0</occluded>
		<bndbox> 
			<xmin>0</xmin>
			<xmax>0</xmax>
			<ymin>0</ymin>
			<ymax>0</ymax>
		</bndbox>
    </object>

**Comment by experiencor** [comment]
You should take a look at my raccoon dataset. Just draw bounding boxes for raccoons using https://github.com/tzutalin/labelImg.

**Comment by jaideep11061982** [comment]
Thanks...this seem to work better than before but now i get so many BBs... Please see attached
![figure_1](https://user-images.githubusercontent.com/38164196/39408917-97bb0c7e-4bfb-11e8-9a48-4ff76189ae37.png)

And loss function dint come down much using inception network

Epoch 00002: val_loss improved from 10.24145 to 10.22068, saving model to yoloimprovement-01-1.00.hdf5
Epoch 3/4
 - 1606s - loss: 10.2370 - val_loss: 10.2148

Where should the problem...
This is my first problem... trying to build more concepts...

**Comment by experiencor** [comment]
Could you send me a few examples of images and annotations?

**Comment by jaideep11061982** [comment]
![receipt11](https://user-images.githubusercontent.com/38164196/39417785-2d276000-4c74-11e8-9e94-78a44abab451.jpeg)
![receipt12](https://user-images.githubusercontent.com/38164196/39417786-2d861e42-4c74-11e8-9a53-630ebbce0ea6.jpeg)
![receipt14](https://user-images.githubusercontent.com/38164196/39417787-2de92f46-4c74-11e8-9b37-a3002b55edf2.jpeg)
![receipt15](https://user-images.githubusercontent.com/38164196/39417788-2e4f1ea0-4c74-11e8-980d-71efbbb375b7.jpeg)
![receipt16](https://user-images.githubusercontent.com/38164196/39417790-2e956392-4c74-11e8-9404-ddaf777966c5.jpeg)
![receipt17](https://user-images.githubusercontent.com/38164196/39417791-2ee240c2-4c74-11e8-9232-e2cc3d862fb1.jpeg)
![receipt18](https://user-images.githubusercontent.com/38164196/39417792-2f7581b6-4c74-11e8-8ffc-5365e2055ad1.jpeg)

**Comment by jaideep11061982** [comment]
In previous commens you can see the dates in picture
below is anotations.xml in pascal voc format ,i think that is what your code understands...


<annotation>
	<folder>images1</folder>
	<filename>receipt2.jpeg</filename>
	<path>C:\work\machine learning\project\images1\receipt2.jpeg</path>
	<source>
		<database>Unknown</database>
	</source>
	<size>
		<width>380</width>
		<height>380</height>
		<depth>3</depth>
	</size>
	<segmented>0</segmented>
	<object>
		<name>Date</name>
		<pose>Unspecified</pose>
		<truncated>1</truncated>
		<difficult>0</difficult>
		<bndbox>
			<xmin>1</xmin>
			<ymin>187</ymin>
			<xmax>168</xmax>
			<ymax>229</ymax>
		</bndbox>
	</object>
	<folder>images1</folder>
	<filename>receipt3.jpeg</filename>
	<path>C:\work\machine learning\project\images1\receipt3.jpeg</path>
	<source>
		<database>Unknown</database>
	</source>
	<size>
		<width>380</width>
		<height>380</height>
		<depth>3</depth>
	</size>
	<segmented>0</segmented>
	<object>
		<name>Date</name>
		<pose>Unspecified</pose>
		<truncated>1</truncated>
		<difficult>0</difficult>
		<bndbox>
			<xmin>213</xmin>
			<ymin>122</ymin>
			<xmax>380</xmax>
			<ymax>162</ymax>
		</bndbox>
	</object>
	<folder>images1</folder>
	<filename>receipt4.jpeg</filename>
	<path>C:\work\machine learning\project\images1\receipt4.jpeg</path>
	<source>
		<database>Unknown</database>
	</source>
	<size>
		<width>380</width>
		<height>380</height>
		<depth>3</depth>
	</size>
	<segmented>0</segmented>
	<object>
		<name>Date</name>
		<pose>Unspecified</pose>
		<truncated>0</truncated>
		<difficult>0</difficult>
		<bndbox>
			<xmin>5</xmin>
			<ymin>260</ymin>
			<xmax>137</xmax>
			<ymax>311</ymax>
		</bndbox>
	</object>
	<folder>images1</folder>
	<filename>receipt5.jpeg</filename>
	<path>C:\work\machine learning\project\images1\receipt5.jpeg</path>
	<source>
		<database>Unknown</database>
	</source>
	<size>
		<width>380</width>
		<height>380</height>
		<depth>3</depth>
	</size>
	<segmented>0</segmented>
	<object>
		<name>Date</name>
		<pose>Unspecified</pose>
		<truncated>1</truncated>
		<difficult>0</difficult>
		<bndbox>
			<xmin>247</xmin>
			<ymin>173</ymin>
			<xmax>380</xmax>
			<ymax>209</ymax>
		</bndbox>
	</object>
	<folder>images1</folder>
	<filename>receipt6 (1).jpeg</filename>
	<path>C:\work\machine learning\project\images1\receipt6 (1).jpeg</path>
	<source>
		<database>Unknown</database>
	</source>
	<size>
		<width>380</width>
		<height>380</height>
		<depth>3</depth>
	</size>
	<segmented>0</segmented>
	<object>
		<name>Date</name>
		<pose>Unspecified</pose>
		<truncated>0</truncated>
		<difficult>0</difficult>
		<bndbox>
			<xmin>82</xmin>
			<ymin>148</ymin>
			<xmax>256</xmax>
			<ymax>209</ymax>
		</bndbox>
	</object>
	<folder>images1</folder>
	<filename>receipt6.jpeg</filename>
	<path>C:\work\machine learning\project\images1\receipt6.jpeg</path>
	<source>
		<database>Unknown</database>
	</source>
	<size>
		<width>380</width>
		<height>380</height>
		<depth>3</depth>
	</size>
	<segmented>0</segmented>
	<object>
		<name>Date</name>
		<pose>Unspecified</pose>
		<truncated>0</truncated>
		<difficult>0</difficult>
		<bndbox>
			<xmin>80</xmin>
			<ymin>154</ymin>
			<xmax>254</xmax>
			<ymax>209</ymax>
		</bndbox>
	</object>
		<folder>images1</folder>
	<filename>receipt7.jpeg</filename>
	<path>C:\work\machine learning\project\images1\receipt7.jpeg</path>
	<source>
		<database>Unknown</database>
	</source>
	<size>
		<width>380</width>
		<height>380</height>
		<depth>3</depth>
	</size>
	<segmented>0</segmented>
	<object>
		<name>Date</name>
		<pose>Unspecified</pose>
		<truncated>1</truncated>
		<difficult>0</difficult>
		<bndbox>
			<xmin>175</xmin>
			<ymin>158</ymin>
			<xmax>380</xmax>
			<ymax>196</ymax>
		</bndbox>
	</object>
	<folder>images1</folder>
	<filename>receipt8.jpeg</filename>
	<path>C:\work\machine learning\project\images1\receipt8.jpeg</path>
	<source>
		<database>Unknown</database>
	</source>
	<size>
		<width>380</width>
		<height>380</height>
		<depth>3</depth>
	</size>
	<segmented>0</segmented>
	<object>
		<name>Date</name>
		<pose>Unspecified</pose>
		<truncated>0</truncated>
		<difficult>0</difficult>
		<bndbox>
			<xmin>118</xmin>
			<ymin>284</ymin>
			<xmax>280</xmax>
			<ymax>354</ymax>
		</bndbox>
	</object>
	<folder>images1</folder>
	<filename>receipt9.jpeg</filename>
	<path>C:\work\machine learning\project\images1\receipt9.jpeg</path>
	<source>
		<database>Unknown</database>
	</source>
	<size>
		<width>380</width>
		<height>380</height>
		<depth>3</depth>
	</size>
	<segmented>0</segmented>
	<object>
		<name>Date</name>
		<pose>Unspecified</pose>
		<truncated>1</truncated>
		<difficult>0</difficult>
		<bndbox>
			<xmin>1</xmin>
			<ymin>236</ymin>
			<xmax>133</xmax>
			<ymax>268</ymax>
		</bndbox>
	</object>
	<folder>images1</folder>
	<filename>receipt10.jpeg</filename>
	<path>C:\work\machine learning\project\images1\receipt10.jpeg</path>
	<source>
		<database>Unknown</database>
	</source>
	<size>
		<width>380</width>
		<height>380</height>
		<depth>3</depth>
	</size>
	<segmented>0</segmented>
	<object>
		<name>Date</name>
		<pose>Unspecified</pose>
		<truncated>0</truncated>
		<difficult>0</difficult>
		<bndbox>
			<xmin>77</xmin>
			<ymin>144</ymin>
			<xmax>364</xmax>
			<ymax>209</ymax>
		</bndbox>
	</object>
	<folder>images1</folder>
	<filename>receipt11.jpeg</filename>
	<path>C:\work\machine learning\project\images1\receipt11.jpeg</path>
	<source>
		<database>Unknown</database>
	</source>
	<size>
		<width>380</width>
		<height>380</height>
		<depth>3</depth>
	</size>
	<segmented>0</segmented>
	<object>
		<name>Date</name>
		<pose>Unspecified</pose>
		<truncated>0</truncated>
		<difficult>0</difficult>
		<bndbox>
			<xmin>12</xmin>
			<ymin>81</ymin>
			<xmax>110</xmax>
			<ymax>111</ymax>
		</bndbox>
	</object>
	<folder>images1</folder>
	<filename>receipt12.jpeg</filename>
	<path>C:\work\machine learning\project\images1\receipt12.jpeg</path>
	<source>
		<database>Unknown</database>
	</source>
	<size>
		<width>380</width>
		<height>380</height>
		<depth>3</depth>
	</size>
	<segmented>0</segmented>
	<object>
		<name>Date</name>
		<pose>Unspecified</pose>
		<truncated>0</truncated>
		<difficult>0</difficult>
		<bndbox>
			<xmin>109</xmin>
			<ymin>247</ymin>
			<xmax>245</xmax>
			<ymax>285</ymax>
		</bndbox>
	</object>
	<folder>images1</folder>
	<filename>receipt13.jpeg</filename>
	<path>C:\work\machine learning\project\images1\receipt13.jpeg</path>
	<source>
		<database>Unknown</database>
	</source>
	<size>
		<width>380</width>
		<height>380</height>
		<depth>3</depth>
	</size>
	<segmented>0</segmented>
	<object>
		<name>Date</name>
		<pose>Unspecified</pose>
		<truncated>1</truncated>
		<difficult>0</difficult>
		<bndbox>
			<xmin>266</xmin>
			<ymin>193</ymin>
			<xmax>380</xmax>
			<ymax>225</ymax>
		</bndbox>
	</object>
	<folder>images1</folder>
	<filename>receipt14.jpeg</filename>
	<path>C:\work\machine learning\project\images1\receipt14.jpeg</path>
	<source>
		<database>Unknown</database>
	</source>
	<size>
		<width>380</width>
		<height>380</height>
		<depth>3</depth>
	</size>
	<segmented>0</segmented>
	<object>
		<name>Date</name>
		<pose>Unspecified</pose>
		<truncated>1</truncated>
		<difficult>0</difficult>
		<bndbox>
			<xmin>298</xmin>
			<ymin>316</ymin>
			<xmax>380</xmax>
			<ymax>346</ymax>
		</bndbox>
	</object>
	<folder>images1</folder>
	<filename>receipt15.jpeg</filename>
	<path>C:\work\machine learning\project\images1\receipt15.jpeg</path>
	<source>
		<database>Unknown</database>
	</source>
	<size>
		<width>380</width>
		<height>380</height>
		<depth>3</depth>
	</size>
	<segmented>0</segmented>
	<object>
		<name>Date</name>
		<pose>Unspecified</pose>
		<truncated>0</truncated>
		<difficult>0</difficult>
		<bndbox>
			<xmin>120</xmin>
			<ymin>55</ymin>
			<xmax>292</xmax>
			<ymax>98</ymax>
		</bndbox>
	</object>
	<folder>images1</folder>
	<filename>receipt16.jpeg</filename>
	<path>C:\work\machine learning\project\images1\receipt16.jpeg</path>
	<source>
		<database>Unknown</database>
	</source>
	<size>
		<width>380</width>
		<height>380</height>
		<depth>3</depth>
	</size>
	<segmented>0</segmented>
	<object>
		<name>Date</name>
		<pose>Unspecified</pose>
		<truncated>0</truncated>
		<difficult>0</difficult>
		<bndbox>
			<xmin>64</xmin>
			<ymin>329</ymin>
			<xmax>188</xmax>
			<ymax>375</ymax>
		</bndbox>
	</object>
   	<folder>images1</folder>
	<filename>receipt17.jpeg</filename>
	<path>C:\work\machine learning\project\images1\receipt17.jpeg</path>
	<source>
		<database>Unknown</database>
	</source>
	<size>
		<width>380</width>
		<height>380</height>
		<depth>3</depth>
	</size>
	<segmented>0</segmented>
	<object>
		<name>Date</name>
		<pose>Unspecified</pose>
		<truncated>0</truncated>
		<difficult>0</difficult>
		<bndbox>
			<xmin>130</xmin>
			<ymin>12</ymin>
			<xmax>358</xmax>
			<ymax>50</ymax>
		</bndbox>
	</object>
	<folder>images1</folder>
	<filename>receipt18.jpeg</filename>
	<path>C:\work\machine learning\project\images1\receipt18.jpeg</path>
	<source>
		<database>Unknown</database>
	</source>
	<size>
		<width>380</width>
		<height>380</height>
		<depth>3</depth>
	</size>
	<segmented>0</segmented>
	<object>
		<name>Date</name>
		<pose>Unspecified</pose>
		<truncated>0</truncated>
		<difficult>0</difficult>
		<bndbox>
			<xmin>112</xmin>
			<ymin>209</ymin>
			<xmax>338</xmax>
			<ymax>255</ymax>
		</bndbox>
	</object>
</annotation>

**Comment by jaideep11061982** [comment]
any modification needed in custom loss which may have been designed wrt to 8-9 classes...
Please suggest the suitable places where i need to make changes as per 1 class

**Comment by tianying107** [comment]
@jaideep11061982 Hello. I got a similar issue here. I trained for one class but the all bounding box conf are lower than 0.1. Do you get some solutions?

**Comment by jaideep11061982** [comment]
I used different annotations tool with that I could solve problem of no
bounding boxes but new problem arose I got now many boxes ,I requested
expriencer for a solution .
I feel some.changes needed in loss procedure as it is for more than one
classes and I have got only one class
